ethpm_cli/registry.py

```diff
@@ -0,0 +1,156 @@
+
+
+def resolve_uri_and_alias(
+    registry_uri: Optional[URI], alias: Optional[str], store_path: Path
+) -> StoredRegistry:
+    if (registry_uri and alias) or (not registry_uri and not alias):
+        raise InstallError("Cannot resolve both an alias and registry uri.")
+
+    registries_and_aliases = get_all_registries_and_aliases(store_path)
+    if alias:
+        registry_uri = lookup_uri_by_alias(alias, registries_and_aliases)
+
+    if not registry_uri or registry_uri not in registries_and_aliases.keys():
+        raise InstallError(
+            f"No registry @ {registry_uri} is available in {store_path}."
+        )
+    return StoredRegistry(registry_uri, False, alias, None)
+
+
+def get_active_registry(store_data: Dict[URI, Any]) -> URI:
+    for registry, data in store_data.items():
+        if data["active"] is True:
+            return registry
+    raise InstallError("Invalid registry store data found.")
+
+
+def lookup_uri_by_alias(alias: str, registries_and_aliases: Dict[URI, str]) -> URI:
+    aliases_and_registries = {v: k for k, v in registries_and_aliases.items()}
+    if alias not in aliases_and_registries:
+        raise InstallError(
+            f"alias: {alias} not available. "
+            f"Available aliases include: {list(aliases_and_registries.keys())}."
+        )
+    return aliases_and_registries[alias]
+
+
+def generate_registry_store(registry_uri: URI, alias: str, store_path: Path) -> None:
+    store_path.touch()
+    init_registry_data = {
+        registry_uri: generate_registry_store_data(registry_uri, alias, activate=True)
+    }
+    write_store_data_to_disk(init_registry_data, store_path)
+
+
+def update_registry_store(registry_uri: URI, alias: str, store_path: Path) -> None:
+    all_registries = get_all_registries_and_aliases(store_path).keys()
+    if registry_uri in all_registries:
+        raise InstallError(f"Registry @ {registry_uri} already stored.")
+    old_store_data = json.loads(store_path.read_text())
+    new_registry_data = generate_registry_store_data(registry_uri, alias)
+    updated_store_data = assoc(old_store_data, registry_uri, new_registry_data)
+    write_store_data_to_disk(updated_store_data, store_path)
+
+
+def get_all_registries_and_aliases(store_path: Path) -> Dict[URI, str]:
+    store_data = json.loads(store_path.read_text())
+    return {uri: store_data[uri]["alias"] for uri in store_data}
+
+
+def write_store_data_to_disk(store_data: Dict[URI, Any], store_path: Path) -> None:
+    with atomic_replace(store_path) as f:
+        f.write(json.dumps(store_data, indent=4, sort_keys=True))
+
+
+@to_dict
+def generate_registry_store_data(
+    registry_uri: URI, alias: str, activate: bool = False
+) -> Iterable[Tuple[str, Any]]:
+    parsed_uri = parse_registry_uri(registry_uri)
+    # todo: support ens in registry uri
+    ens = None
+    yield "ens", ens
+    yield "address", parsed_uri.address
+    yield "alias", alias
+    yield "active", activate
```

Comment:
Since there's no logic in this function to exclude any keys you can just construct and return the dictionary without using the `to_dict` approach.  probably a little simpler to deal with.